File: MAP/Interpretor2/src/Model/MyStack.java
package Model;

import java.util.Stack;

public class MyStack<T> implements MyIStack<T> {
	Stack<T> stack;
	
public MyStack()
{
	this.stack=new Stack <T>() ;
}
public void push(T elem)
{
	this.stack.push(elem);
}
public T pop()
{
	return this.stack.pop();
}
public boolean isEmpty()
{
	return this.stack.isEmpty();
}
@Override
public String toString(){
	String str="";
	for(T e:this.stack)
	{
		str=str+e.toString()+" ";
	}
	return str;
}


}

File: Interpreter C#/Interpreter/MyStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Interpreter
{
    namespace Model
    {
        class MyStack<T> : MyIStack<T>
        {
            Stack<T> stack;

          public MyStack()
            {
                this.stack = new Stack<T>();
            }

            T MyIStack<T>.pop()
            {
                return this.stack.Pop();
            }

            void MyIStack<T>.push(T v)
            {
                this.stack.Push(v);
            }

            public override string ToString()
            {
                string str=" ";
                foreach(T i in this.stack)
                {
                    str = str + i.ToString();
                }
                return str;
            }

            Boolean MyIStack<T>.isEmpty()
            {
                return this.stack.Count() == 0;
            }
        }
    }
}
